DataProcession.py

```diff
@@ -0,0 +1,68 @@
+import re
+from ShowResult import ShowReult
+
+class DataProcessing:
+    """This class calculate the different reports as per format"""
+    def max_min_calculate(self, data):
+        """Calculate the highest, lowest temprature resp. and most humidity 
+
+        Args:
+            data ([list]): [required data on which we do processing ]
+
+        Returns:
+            [Results]: [result storage class]
+        """
+        self.index1 = data.names.index('Max TemperatureC')
+        self.index2 = data.names.index('Min TemperatureC')
+        self.index3 = data.names.index('Max Humidity')
+
+        max_temprature = max([(sublist[self.index1], sublist[0]) for sublist in data.values if sublist[self.index1] is not None])
+        min_temprature = min([(sublist[self.index2], sublist[0]) for sublist in data.values if sublist[self.index2] is not None])
+        max_humidity = max([(sublist[self.index3], sublist[0]) for sublist in data.values if sublist[self.index3] is not None])
+        
+        return ShowReult(max_temprature, min_temprature, max_humidity)
+
+    def average_values_calculate(self, data):
+        """Calculate the highest, lowest avarage temprature resp. and average mean humidity 
+
+        Args:
+            data ([list]): [required data on which we do processing ]
+
+        Returns:
+            [Results]: [result storage class]
+        """
+
+        self.index1 = data.names.index('Max TemperatureC')
```

Comment:
Rather than getting this value in each method. Why can't we move this to __init__() so we just have to call it once and then use in different methods using `self`